src/App.js

```diff
@@ -2,12 +2,66 @@ import React from 'react';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import GoodsList from './GoodsList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.PureComponent {
+  state = {
+    dataLoaded: false,
+    arrayOfGoods: [],
+  };
+
+  loadAllGoods = () => {
+    this.setState(getAll().then(array => this.setState({
+      arrayOfGoods: array,
+      dataLoaded: true,
+    })));
```

Comment:
are you sure, that it's good idea to call setState inside the setState? it looks weird. get rid of outer setState call